tests/test_tools.py

```diff
@@ -853,24 +813,37 @@ def test_bool_or_int_to_str(inp, exp):
 
 
 @pytest.mark.parametrize('inp, exp', [
-        (42, 42),
+        (42, slice(42, 43)),
         (None, slice(None, None, None)),
-        ('42', 42),
-        ('-42', -42),
+        ('42', slice(42, 43)),
+        ('-42', slice(-42, -41)),
         ('1:2:3', slice(1, 2, 3)),
         ('1::3', slice(1, None, 3)),
         (':', slice(None, None, None)),
         ('1:', slice(1, None, None)),
         ('[1:2:3]', slice(1, 2, 3)),
         ('(1:2:3)', slice(1, 2, 3)),
-        ('r', False),
-        ('r:11', False),
         ])
-def test_ensure_int_or_slice(inp, exp):
-    obs = ensure_int_or_slice(inp)
+def test_ensure_slice(inp, exp):
+    obs = ensure_slice(inp)
     assert exp == obs
 
 
+@pytest.mark.parametrize('inp', [
+    '42.3',
+    '3:asd5:1',
+    'test',
+    '6.53:100:5',
+    '4:-',
+    '2:15-:3',
+    '50:-:666',
+    object(),
+    ])
+def test_ensure_slice_invalid(inp):
+    obs = ensure_slice(inp)
+    assert obs is None
```

Comment:
i wanted to ask you about this, the previous `ensure_int_or_slice` returned False, i prefer raising and error too.